src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +637,7210 @@ ActorProfile Boss_07_Profile = {
+    MAJORAS_WRATH_BODYPART_TORSO,                // MAJORAS_WRATH_BODYPART_PELVIS
+    MAJORAS_WRATH_BODYPART_TORSO,                // MAJORAS_WRATH_BODYPART_LEFT_UPPER_ARM
+    MAJORAS_WRATH_BODYPART_LEFT_UPPER_ARM,       // MAJORAS_WRATH_BODYPART_LEFT_LOWER_ARM_ROOT
+    MAJORAS_WRATH_BODYPART_LEFT_LOWER_ARM_ROOT,  // MAJORAS_WRATH_BODYPART_LEFT_FOREARM
+    MAJORAS_WRATH_BODYPART_TORSO,                // MAJORAS_WRATH_BODYPART_RIGHT_UPPER_ARM
+    MAJORAS_WRATH_BODYPART_RIGHT_UPPER_ARM,      // MAJORAS_WRATH_BODYPART_RIGHT_LOWER_ARM_ROOT
+    MAJORAS_WRATH_BODYPART_RIGHT_LOWER_ARM_ROOT, // MAJORAS_WRATH_BODYPART_RIGHT_FOREARM
+    MAJORAS_WRATH_BODYPART_PELVIS,               // MAJORAS_WRATH_BODYPART_RIGHT_THIGH
+    MAJORAS_WRATH_BODYPART_RIGHT_THIGH,          // MAJORAS_WRATH_BODYPART_RIGHT_SHIN
+    MAJORAS_WRATH_BODYPART_RIGHT_SHIN,           // MAJORAS_WRATH_BODYPART_RIGHT_FOOT
+    MAJORAS_WRATH_BODYPART_PELVIS,               // MAJORAS_WRATH_BODYPART_LEFT_THIGH
+    MAJORAS_WRATH_BODYPART_LEFT_THIGH,           // MAJORAS_WRATH_BODYPART_LEFT_SHIN
+    MAJORAS_WRATH_BODYPART_LEFT_SHIN             // MAJORAS_WRATH_BODYPART_LEFT_FOOT
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_07/func_80A06500.s")
+static u8 sShadowSizes[MAJORAS_WRATH_BODYPART_MAX] = {
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_HEAD
+    MAJORAS_WRATH_SHADOW_SIZE_EXTRA_LARGE, // MAJORAS_WRATH_BODYPART_TORSO
+    MAJORAS_WRATH_SHADOW_SIZE_EXTRA_LARGE, // MAJORAS_WRATH_BODYPART_PELVIS
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_LEFT_UPPER_ARM
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_LEFT_LOWER_ARM_ROOT
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_LEFT_FOREARM
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_RIGHT_UPPER_ARM
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_RIGHT_LOWER_ARM_ROOT
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_RIGHT_FOREARM
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_RIGHT_THIGH
+    MAJORAS_WRATH_SHADOW_SIZE_MEDIUM,      // MAJORAS_WRATH_BODYPART_RIGHT_SHIN
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_RIGHT_FOOT
+    MAJORAS_WRATH_SHADOW_SIZE_LARGE,       // MAJORAS_WRATH_BODYPART_LEFT_THIGH
+    MAJORAS_WRATH_SHADOW_SIZE_MEDIUM,      // MAJORAS_WRATH_BODYPART_LEFT_SHIN
+    MAJORAS_WRATH_SHADOW_SIZE_SMALL,       // MAJORAS_WRATH_BODYPART_LEFT_FOOT
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_07/func_80A06990.s")
+void Boss07_Wrath_FillShadowTex(Boss07* this, u8* tex, f32 weight) {
+    s32 index;
+    s32 i;
+    s32 baseX;
+    s32 baseY;
+    s32 x;
+    s32 y = 0;
+    s32 addY;
+    Vec3f lerp;
+    Vec3f pos;
+    Vec3f startVec;
+
+    for (i = 0; i < MAJORAS_WRATH_BODYPART_MAX; i++) {
+        // TODO: match with a continue
```

Comment:
It's discussed in more detail here: https://discord.com/channels/688807550715560050/688851976746041391/1155281303831314483

To summarize, the current way of writing it is vertically compact, but it requires an assignment within an if-statement and also puts an extra level of indentation into the entire function. I know I can match the usages in Odolwa, Goht, and Majora using a `continue`, and I swear I *did* match EnSda using a continue as well, but I never saved it anywhere, lost it in a revert, and could never get it matching again. I didn't want to have that slow me down on Goht (particularly because, at the time, Majora was reserved by someone else, so I didn't feel comfortable going in and modifying it to match the style I wanted to standardize on), so I just left comments instead.

Given that all the dynamic shadow actors have a comment like this at this point, we can either leave this comment here or just remove it everywhere else. It wouldn't really make sense to remove it here and keep it in all the other actors.

Reply:
My personal thought is to leave the todo for now. Lets swing back around to it after this PR.